### ISSUE TITLE
Fullstack practice project review

### ISSUE THREAD

**Issue by ShootingStar91** [Ammar-daham/library-system#1]
# Fullstack practice project review

This is the review of your fullstack practice project for the https://fullstackopen.com course. It may have suggestions for improvement, but you do not need to implement those in order to get the credits for the project.

### What I did

+ Signed up and logged in
+ Added author and edited it's email, then deleted it
+ Added book and edited it's name
+ Added an author
+ Added a genre
+ Tested form validations
+ Logged out

### User experience

+ A nice look and feel, easy to sign up and forms look nice. Good notification and a redirect after signing in
+ The navigation bar links work only if clicked on the text, which is confusing, because the highlight and animation appears on a larger area
+ All the functionalities work well, and validations exist
+ The UI looks nice. It's reflexive and looks very usable on mobile too, great!
+ The extra menu in the bottom is redundant, as everything is nicely accessible on top.
+ The "Update" and "Delete" buttons in the genres and author views need some gap between them, also they don't need to be so wide

### Code

+ Good use of technologies like Docker, Typescript, Redux
+ `!menu ? setMenu(true) : setMenu(false)` could also just be `setMenu(!menu)` :)
+ The notification produces a lot of duplicate code. It could be handled via redux to replace all these
```
setSuccessMessage(`Thank you, you have successfully deleted the book ${id}`)
			setTimeout(() => {
				setSuccessMessage('')
				setErrorMessage('')
				window.location.href = '/'; 
			}, 3000)
```

With something like 

```dispatch(success(`Thank you, you have successfully deleted the book ${id}`))```

+ The code in general is easy to read and organized in logical folders and files, in both frontend and backend.
+ In backend, it would be a bit more readable by using [express-async-errors](https://www.npmjs.com/package/express-async-errors?activeTab=readme) to remove the try-catch clutter in the controller files

In summary, the application provides multiple features that all worked without problems. Fields have validations and a nice notification. The interface is clear and simple to use. Good job!
